tensorflow/compiler/xla/client/xla_builder.cc

```diff
@@ -766,15 +766,17 @@ XlaOp XlaBuilder::BroadcastInDim(
     TF_ASSIGN_OR_RETURN(auto output_shape,
                         ShapeUtil::MakeValidatedShape(
                             operand_shape->element_type(), out_dim_size));
-    if (operand_shape->rank() != broadcast_dimensions.size()) {
+    tensorflow::int64 broadcast_dimensions_size = broadcast_dimensions.size();
+    if (operand_shape->rank() != broadcast_dimensions_size) {
       return InvalidArgument(
           "Size of broadcast_dimensions has to match operand's rank; operand "
           "rank: %lld, size of broadcast_dimensions %u.",
           operand_shape->rank(), broadcast_dimensions.size());
     }
-    for (int i = 0; i < broadcast_dimensions.size(); i++) {
+    for (int i = 0, end = broadcast_dimensions.size(); i < end; i++) {
```

Comment:
```suggestion
    for (int i = 0; i < broadcast_rank; i++) {
```

(we already have the local)